### DISCUSSION TITLE
Make electric bolt animation speed customizable

### DISCUSSION THREAD

**Discussion by mevitar** [Phobos-developers/Phobos#1125] [Ideas and feedback]
### Description

IsElectricBolt=yes effects creates line arcs that appear to be spinning for several seconds. This request is to make the "spin" speed and/or duration customizable.

### INI code

```ini
[SOMEWEAPONTYPE]
Bolt.SpinTime=
Bolt.SpinSpeed=

or just
Bolt.Duration=
Bolt.Speed=
```

### Checklist

- [X] The suggestion is **NOT** related to functionality of Ares, HAres, Kratos or any other YR engine extension.
- [X] The suggested feature / enhancement is **NOT** implemented in Ares/Phobos yet and the existing possibilites are not viable or good enough to achieve the wanted result.
- [X] I agree to elaborate the details if requested and provide thorough testing if the feature is implemented.
- [X] I added a very descriptive title to this topic.
- [X] I used the GitHub search and read the topic list to find a similar topic and didn't find it.
- [X] I thought out how the feature / enhancement should work and described it in the most detailed and clear way.

**Comment by Starkku** [comment]
This is probably doable, barring some conflict with Ares electric bolt additions.

The way electric bolts work is that they have lifetime value that always starts at 65536, and a random seed number that goes from 0-256 which is used to randomize the blot (line) drawing. Every frame when bolts are rendered it checks if the lifetime value is non-zero, draws the lines if so and then increments the random seed value by 1. And on every game frame the lifetime is divided by 2 (it is stored as integer so once it goes below 1 it becomes 0). I believe what you're essentially asking for is customizable lifetime and frame rate (how many game frames it takes to increment the random seed) for the electric bolts.

**Comment by mevitar** [comment]
I was hoping to do a static bolt (like the lightning-from-the-sky kind of bolt) instead of the current one (dancing-all-over-the place tesla bolt), and modify its duration. From your description it seems that there is no other way than doing it like what you said, since the game doesn't do a clean "time remaining - time elapsed" substraction.

But i guess that the lifetime would be customized through changing the initial lifetime value, right? I don't see how you could do it any other way.
